### DISCUSSION TITLE
Moving Shell-Elements to an inactive subdomain (error in MooseVariableBase.C, line 145)

### DISCUSSION THREAD

**Discussion by jmeier** [idaholab/moose#28201] [Q&A Modules: Solid mechanics] [answered]
Dear Moose-Community,

I have a 3D mesh with some volume-elements in one subdomain and some shell-elements in another subdomain. The moment I try to move the shell-elements into an inactive domain (added by `add_subdomain_names`), I get the following error:

```
Assertion `_is_lower_d == _mesh.isLowerD(*it)' failed
A user should not specify a mix of lower-dimensional and higher-dimensional blocks for a variable
at /home/user/projects/moose/framework/src/variables/MooseVariableBase.C, line 145
```

The inactive domain I tried to move the shell-elements to was previously empty and never used before. I only try to move the shell elements of one subdomain (I do not try to mix element types in one inactive subdomain).

Is it possible to move shell elements to an inactive domain, if the mesh consists also of volume elements?

Thank you for your help.
Jörg

**Comment by GiudGiud** [comment, marked as answer]
Hello

We'll have to change the code a little bit because we should allow this.
This is more about the elements being 2D in a 3D mesh than being of the "XYZShell" type but I understand what you mean.

Guillaume

**Reply by GiudGiud** [reply]
let's create an issue

**Comment by jmeier** [comment]
It seems that the error I am experiencing is not primarily related to assigning shell elements to an inactive subdomain. I am therefore closed the issue #28204 as resolved. To #28204 I attached a test model which shows the moving of volume and shell elements with a TimedSubdomainModifier without generating an error.

**Reply by GiudGiud** [reply]
it should be tied to dimensions of the elements (with dim=2 and dim=3 elements at the same time for a variable being the problem).
If that is the case then this is expected.

I think we could change this if needed
